AAInstrumentation/unique_hashes.py

```diff
@@ -0,0 +1,67 @@
+from core import AAInstrumentationDriver, register_arguments
+from dataclasses import dataclass, field
+import sys
+from pathlib import Path
+from typing import Dict, List
+import random
+
+
+@dataclass
+class UniqueHashesDriver(AAInstrumentationDriver):
+    def get_num_unique_hashes(
+        self,
+        file_name: Path,
+        num_candidates: int,
+        num_runs: int,
+    ) -> int:
+        population = []
+        distinct_hashes = set()
+
+        for _ in range(num_runs):
+            curr_list = tuple(random.randint(0, 1) for _ in range(num_candidates))
+            index_list = []
+            for j, val in enumerate(curr_list):
+                if val:
+                    index_list.append(j)
+            population.append(tuple(index_list))
```

Comment:
this loop essentially creates a random bit vector and converts it into an "index vector", right?

If so, would it make sense to factor it into a separate function and give it an appropriate name?

Reply:
Good point, this can be resued for an optimizers. Moved parts to core.